### PR TITLE
chore(telemetry): allow sharing tags in a file and read it only when things change

### PR DESCRIPTION
**Pull request by jamo**
Allow reading telemetry data from a file

### CODE REVIEW

**Comment by jamo** [comment]
I had stale reads / edits with out the sleeps. Might be just my `zfs` fs but the multi-second sleep was the only way I got it to be reliable.

**Comment by jamo** [comment]
This has been tested manually and is working well.

File: packages/gatsby-telemetry/src/__tests__/telemetry.ts [hunk 1]
```diff
@@ -57,4 +61,119 @@ describe(`Telemetry`, () => {
...
+      expect(t.lastEnvTagsFromFileTime).toBe(0)
+
+      // create file and write initial data
+      const filePath = path.join(fs.realpathSync(os.tmpdir()), uuidv4())
+      console.log(filePath)
+      process.env.GATSBY_TELEMETRY_METADATA_PATH = filePath
+
+      fs.writeFileSync(filePath, JSON.stringify({ componentId: `test` }))
+      await new Promise(resolve => {
+        setTimeout(resolve, 2000)
```
Comment (wardpeet) [inline packages/gatsby-telemetry/src/__tests__/telemetry.ts:83]: Why is this necessary?

**Review by wardpeet — APPROVED** [review]
I'm not sure what this does but code looks fine

File: packages/gatsby-telemetry/src/__tests__/telemetry.ts (hunk 1 above)
Comment (jamo) [reply packages/gatsby-telemetry/src/__tests__/telemetry.ts:83]: yea, the only way I got it working was to force sleep, I did spend a lot of time trying lower level APIs to ensure I don't get stale reads.

**Comment by vladar** [comment]
Published in `gatsby-telemetry@1.8.1`
